volume/volume.go

```diff
@@ -120,6 +125,27 @@ type MountPoint struct {
 
 	// Sepc is a copy of the API request that created this mount.
 	Spec mounttypes.Mount
+
+	// Track usage of this mountpoint
+	// Specicially needed for containers which are running and calls to `docker cp`
+	// because both these actions require mounting the volumes.
+	active activeMount
+}
+
+// Cleanup frees resources used by the mountpoint
+func (m *MountPoint) Cleanup() error {
+	if m.Volume == nil || m.ID == "" {
+		return nil
+	}
+
+	if err := m.Volume.Unmount(m.ID); err != nil {
+		return errors.Wrapf(err, "error unmounting volume %s", m.Volume.Name())
+	}
+	atomic.AddInt32(&m.active.count, -1)
+	if active := atomic.LoadInt32(&m.active.count); active == 0 {
+		m.ID = ""
+	}
```

Comment:
I believe a race is still possible here. The value of `m.active.count` could change after `LoadInt32()` returns. Which would mean the ID in setup would be erased.